x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/AbstractScriptFieldScript.java

```diff
@@ -21,23 +24,19 @@
  */
 public abstract class AbstractScriptFieldScript {
     private final Map<String, Object> params;
-    private final LeafReaderContext ctx;
-    private final SourceLookup source;
-    private final LeafDocLookup fieldData;
+    private final LeafSearchLookup leafSearchLookup;
 
-    public AbstractScriptFieldScript(Map<String, Object> params, SourceLookup source, DocLookup fieldData, LeafReaderContext ctx) {
+    public AbstractScriptFieldScript(Map<String, Object> params, SearchLookup searchLookup, LeafReaderContext ctx) {
+        this.leafSearchLookup = searchLookup.getLeafSearchLookup(ctx);
+        //TODO how do other scripts get stored fields exposed? Through asMap? I don't see any getters for them.
```

Comment:
Yeah, they have them exposed in *weird* ways that I don't feel comfortable copying. I'd prefer to make a getter for them.